src/mlpack/tests/recurrent_network_test.cpp

```diff
@@ -635,9 +652,18 @@ void DistractedSequenceRecallTestNetwork()
  * Train the specified networks on the Derek D. Monner's distracted sequence
  * recall task.
  */
-BOOST_AUTO_TEST_CASE(DistractedSequenceRecallTest)
+BOOST_AUTO_TEST_CASE(LSTMDistractedSequenceRecallTest)
+{
+  DistractedSequenceRecallTestNetwork<LSTM<>>();
+}
+
+/**
+ * Train the specified networks on the Derek D. Monner's distracted sequence
+ * recall task.
+ */
+BOOST_AUTO_TEST_CASE(GRUDistractedSequenceRecallTest)
 {
-  DistractedSequenceRecallTestNetwork();
+  DistractedSequenceRecallTestNetwork<GRU<>>();
```

Comment:
I like the idea, but I'm not sure the test would provide much insight about the correctness of the GRU or LSTM implementation. Like in some situations LSTM might converge faster as the GRU layer.